rpcs3/Emu/Cell/Modules/cellPad.cpp

```diff
@@ -312,30 +312,36 @@ error_code cellPadGetData(u32 port_no, vm::ptr<CellPadData> data)
 		data->button[CELL_PAD_BTN_OFFSET_ANALOG_RIGHT_Y] = pad->m_analog_right_y;
 		data->button[CELL_PAD_BTN_OFFSET_ANALOG_LEFT_X] = pad->m_analog_left_x;
 		data->button[CELL_PAD_BTN_OFFSET_ANALOG_LEFT_Y] = pad->m_analog_left_y;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_RIGHT] = pad->m_press_right;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_LEFT] = pad->m_press_left;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_UP] = pad->m_press_up;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_DOWN] = pad->m_press_down;
-	}
 
-	if (data->len >= CELL_PAD_LEN_CHANGE_PRESS_ON)
-	{
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_TRIANGLE] = pad->m_press_triangle;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_CIRCLE] = pad->m_press_circle;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_CROSS] = pad->m_press_cross;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_SQUARE] = pad->m_press_square;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_L1] = pad->m_press_L1;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_L2] = pad->m_press_L2;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_R1] = pad->m_press_R1;
-		data->button[CELL_PAD_BTN_OFFSET_PRESS_R2] = pad->m_press_R2;
-	}
+		if (pad->m_port_setting & CELL_PAD_SETTING_PRESS_ON)
```

Comment:
Are we missing some length checks? Or is the check against CELL_PAD_LEN_NO_CHANGE enough? There is a check against CELL_PAD_LEN_CHANGE_PRESS_ON that was removed here, I'm just wondering if this could now overflow.

Reply:
Yes, CELL_PAD_LEN_NO_CHANGE check is enough. the test shows that its modified regardless of the option or the setting if there was any data. which is why I removed a comment saying that relhw only writes up to len.

Reply:
Thanks